Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -26,21 +41,46 @@ public struct DiagnosticsFormatter {
   /// Whether to colorize formatted diagnostics.
   public let colorize: Bool
 
-  public init(contextSize: Int = 2, colorize: Bool = false) {
+  public init(contextSize: Int = defaultContextSize, colorize: Bool = false) {
     self.contextSize = contextSize
     self.colorize = colorize
   }
 
+  public init(contextSize: Int = defaultContextSize, colorMode: ColorMode) {
+    var colorize: Bool
+
+    switch colorMode {
+    case .auto:
+      colorize = TerminalHelper.isConnectedToTerminal
```

Comment:
This is the bit that I think we should life into `swift-parser-cli`, because we're trying to keep the `SwiftDiagnostics` library (and everything that gets linked into the compiler proper) with minimal dependencies.